### PR TITLE
eliminates typing errors on existing code

### PR DESCRIPTION
**Pull request by ca0v**
1. lang._mixin -> lang.mixin (Possible typo or I'm missed a memo?)
2. Existing code uses query.onclick(callback) but typings complained
3. topic.publish(topic) is valid but was saying a second param was required.
4. require({configuration}) is valid, yes?  I've been doing it for years now so I hope so!

### CODE REVIEW

**Comment by kitsonk** [comment]
Sorry it has taken me so long to look...

File: dojo/1.11/_base.d.ts [hunk 1]
```diff
@@ -1275,7 +1275,7 @@ declare namespace dojo {
 			/**
 			 * Copies/adds all properties of one or more sources to dest; returns dest.
 			 */
-			_mixin<T, U>(dest: T, source: U, copyFunc?: (s: any) => any): T & U;
+			mixin<T, U>(dest: T, source: U, copyFunc?: (s: any) => any): T & U;
```
Comment (kitsonk) [inline dojo/1.11/_base.d.ts:1278]: [`_mixin` is correctly typed](https://github.com/dojo/dojo/blob/06203f380069d21b48b606577ea3d507cbfd381b/_base/lang.js#L68) but what is missing is:

``` typescript
/**
 * Copies/adds all properties of one or more sources to dest; returns dest.
 */
mixin<T, U>(dest: T, sources: U): T & U;
mixin<T, U, V>(dest: T, source1: U, source2: V): T & U & V;
mixin<T, U, V, W>(dest: T, source1: U, source2: V, source3: W): T & U & V & W;
mixin<T, U, V, W, X>(dest: T, source1: U, source2: V, source3: W, source4: X): T & U & V & W & X;
mixin<T, U, V, W, X, Y>(dest: T, source1: U, source2: V, source3: W, source4: X, source5: Y): T & U & V & W & X & Y;
mixin<T, U, V, W, X, Y, Z>(dest: T, source1: U, source2: V, source3: W, source4: X, source5: Y, source6: Z): T & U & V & W & X & Y & Z;
mixin<T, U>(dest: T, ...sources: U[]): T & U;
```

File: dojo/1.11/dojo.d.ts [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /// <reference path="index.d.ts" />
-/// <reference path="../../doh/1.11/doh" />
+/// <reference path="../../doh/1.11/doh.d.ts" />
```
Comment (kitsonk) [inline dojo/1.11/dojo.d.ts:2]: Typescript should resolve `doh` or `doh.d.ts`... I guess we should be consistent though.

File: dojo/1.11/_base.d.ts (hunk 1 above)
Comment (kitsonk) [inline dojo/1.11/_base.d.ts:1278]: Actually, I have done that via e983eb0030a4c31c176a49d6129a662dbc5ceac4

File: dojo/1.11/modules.d.ts [hunk 3]
```diff
@@ -731,3 +731,14 @@ declare module 'dojo/window' {
 	const window: dojo.WindowModule;
 	export = window;
 }
+
+declare module 'dojo/Stateful' {
```
Comment (kitsonk) [inline dojo/1.11/modules.d.ts:735]: I added this to fix another issue just now in a6b105a1a738b596eab3d3ebcd937c2f41072ef1

File: dojo/1.11/store.d.ts [hunk 4]
```diff
@@ -261,7 +261,19 @@ declare namespace dojo {
 			/**
 			 * This is a basic in-memory object store. It implements dojo/store/api/Store.
 			 */
-			new <T extends Object>(options?: MemoryOptions<T>): Memory<T>;
+			new <T extends Object>(options?: MemoryOptions<T>): T & Memory<T>;
```
Comment (kitsonk) [inline dojo/1.11/store.d.ts:264]: I am pretty sure this isn't what you would want.

File: dojo/1.11/store.d.ts [hunk 5]
```diff
@@ -261,7 +261,19 @@ declare namespace dojo {
...
+		}
+
+		interface Observable<T> extends dojo.Stateful {
```
Comment (kitsonk) [inline dojo/1.11/store.d.ts:267]: Observable is a bit more complicated than this expresses.  I think I got it properly typed in 020de14d6d8bebabe2340eed2a863baedc873f9f.

File: dojo/1.11/dojo.d.ts (hunk 2 above)
Comment (dylans) [reply dojo/1.11/dojo.d.ts:2]: This was resolved in 5f6abdb.

**Comment by dylans** [comment]
Given how much of this has landed in other commits, I've extracted the remaining items as https://github.com/dojo/typings/pull/79 . One exception is sort, which this PR had added to QueryResults, but it's not actually part of the QueryResults module, so I left that alone for now.

**Comment by dylans** [comment]
Also split typings for dijit/form/Select into a separate issue (https://github.com/dojo/typings/issues/80) as it needs its interface defined before it should be added to the modules list.

**Pull request closed without merging**